src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -233,7 +234,8 @@ public override void ToExpression(StringBuilder sb, FormulaValueSerializerSettin
 
                 flag = false;
 
-                sb.Append($"'{CharacterUtils.Escape(field.Name)}':");
+                sb.Append(IdentToken.MakeValidIdentifier(field.Name));
```

Comment:
>IdentToken.MakeValidIdentifier [](http://example.com/codeflow?start=26&length=30)

Should this be called from a helper (CharacterUtils?) so we have a centralized place to handle these scenarios?

Reply:
CharacterUtils is about escaping string values. IdentToken is the place where we deal with identifiers (which is the case for record property names). I agree it's confusing, we can consider unifying them all in a single place in the future.

Reply:
Bonus that IdentToken.MakeValidIdentifier is public, so derived classes can call it.